### PR TITLE
Add remaining fields to drainagestrategy

### PR DESCRIPTION
**Pull request by petterwildhagen**
Add remaining fields to drainage strategy:
 ProductionProfileWater
 ProductionProfileWaterInjection
 FuelFlaringAndLosses
 NetSalesGas
 Co2Emissions

Introduce new base class TimeSeriesMass to handle time series of masses (Co2 emissions)

Updated test data generator

Updated unit tests

### CODE REVIEW

File: backend/api/Models/DrainageStrategy.cs [hunk 1]
```diff
@@ -1,6 +1,8 @@
 
 using System.ComponentModel.DataAnnotations.Schema;
 
+using Microsoft.EntityFrameworkCore.Metadata.Internal;
```
Comment (valentin-krasontovitsch) [inline backend/api/Models/DrainageStrategy.cs:4]: I'm so sorry, i'm trying a bit of a thorough review, and you're receiving all these possibly silly comments / questions, so please bear with me (hopefully just a bit longer) - how come you added this second using statement? based on the other added lines in this file, i don't see anything "conceptually new" that would require an additional `using` added, and indeed in a quick test the code builds with that using statement commented out.

File: backend/tests/Services/ProjectService.cs [hunk 2]
```diff
@@ -46,86 +46,103 @@ public void GetProject()
...
+        void compareYearValues<T>(TimeSeriesBase<T> expected, TimeSeriesBase<T> actual)
+        {
+            if (expected == null || actual == null)
+            {
+                Assert.Equal(expected, null);
+                Assert.Equal(actual, null);
+            }
+            else
+            {
+                var yearValuePairsXY = expected.YearValues.OrderBy(v => v.Year).Zip(actual.YearValues.OrderBy(v => v.Year));
```
Comment (valentin-krasontovitsch) [inline backend/tests/Services/ProjectService.cs:145]: does it make sense to compare the length of both lists? based on the docs it seems like if one of the list is a proper sublist of the other, this test will not fail. or is that what we want?

**Review by valentin-krasontovitsch — CHANGES_REQUESTED** [review]
sorry, quite a few comments, but hopefully just minor questions : )

File: backend/api/Context/TestDataBuilder.cs [hunk 3]
```diff
@@ -97,3 +127,68 @@ public ProductionProfileGasBuilder WithYearValue(int y, double v)
...
+    }
+}
+
+public class Co2EmissionsBuilder : Co2Emissions
+{
+    public Co2EmissionsBuilder()
+    {
+        YearValues = new List<YearValue<double>>();
+    }
+    public Co2EmissionsBuilder WithYearValue(int y, double v)
```
[the remark this answers is not in the document]
Comment (petterwildhagen) [reply backend/api/Context/TestDataBuilder.cs:189]: Fixed

File: backend/api/Models/TimeSeries.cs [hunk 4]
```diff
@@ -11,13 +12,22 @@ public abstract class TimeSeriesBase<T>
...
     public class TimeSeriesVolume<T> : TimeSeriesBase<T>
     {
-        public Unit Unit { get; set; }
+        public VolumeUnit Unit { get; set; }
     }
-    public enum Unit
+    public enum VolumeUnit
     {
         SM3,
         BBL
```
[the remark this answers is not in the document]
Comment (petterwildhagen) [reply backend/api/Models/TimeSeries.cs:20]: The code should be self documenting so ideally no explanation or inline comments. The domain model should be a description of the domain. So leave as is for now.

File: backend/api/Models/DrainageStrategy.cs (hunk 1 above)
Comment (petterwildhagen) [reply backend/api/Models/DrainageStrategy.cs:4]: See line 12, this is needed for the foreign key statement

File: backend/tests/Services/ProjectService.cs (hunk 2 above)
Comment (petterwildhagen) [reply backend/tests/Services/ProjectService.cs:145]: Yes, it makes sense. Fixed.

File: backend/api/Context/TestDataGenerator.cs [hunk 5]
```diff
@@ -44,20 +44,56 @@ public static ProjectsBuilder initialize()
...
+                        Unit = VolumeUnit.SM3
+                    }
+                        .WithYearValue(2030, 11.23)
+                        .WithYearValue(2031, 11.45)
+                        .WithYearValue(2032, 11.56)
+                    )
+                    .WithCo2Emissions(new Co2EmissionsBuilder()
+                    {
+                        Unit = MassUnit.TON
+                    })
```
[the remark this answers is not in the document]
Comment (petterwildhagen) [reply backend/api/Context/TestDataGenerator.cs:96]: Ah - I forgot to add year values. Fixed.

File: backend/api/Models/Co2Emissions.cs [hunk 6]
```diff
@@ -0,0 +1,10 @@
+using System.ComponentModel.DataAnnotations.Schema;
+
+namespace api.Models
+{
+    public class Co2Emissions : TimeSeriesMass<double>
+    {
+        [ForeignKey("DrainageStrategy.Id")]
```
[the remark this answers is not in the document]
Comment (petterwildhagen) [reply backend/api/Models/Co2Emissions.cs:7]: The foreign key specifies which entity is the dependent one.

**Review by valentin-krasontovitsch — APPROVED** [review]
thanks for the answers and fixes!

**Review by valentin-krasontovitsch — APPROVED** [review]

**Pull request merged**